cloudevents-sdk-tide/src/server_request.rs

```diff
@@ -0,0 +1,291 @@
+use super::headers;
+use bytes::{Bytes, BytesMut};
+use cloudevents::event::SpecVersion;
+use cloudevents::message::{
+    BinaryDeserializer, BinarySerializer, Encoding, MessageAttributeValue, MessageDeserializer,
+    Result, StructuredDeserializer, StructuredSerializer,
+};
+use cloudevents::{message, Event};
+use std::collections::HashMap;
+use std::convert::TryFrom;
+use tide::{Error, Request};
+
+/// Wrapper for [`Request`] that implements [`MessageDeserializer`] trait.
+pub struct RequestDeserializer {
+    headers: HashMap<String, String>,
+    body: Bytes,
+}
+
+impl RequestDeserializer {
+    pub fn new(headers: HashMap<String, String>, body: Bytes) -> RequestDeserializer {
+        RequestDeserializer { headers, body }
+    }
+}
+
+impl<'a> BinaryDeserializer for RequestDeserializer {
+    fn deserialize_binary<R: Sized, V: BinarySerializer<R>>(self, mut visitor: V) -> Result<R> {
+        if self.encoding() != Encoding::BINARY {
+            return Err(message::Error::WrongEncoding {});
+        }
+
+        let versionheader = match self.headers.get("ce-specversion") {
+            Some(s) => s.as_str(),
+            None => "",
```

Comment:
No spec version should fail

Reply:
Thanks for this @slinkydeveloper I will take a look and update